src/Listener/RelatedModelsListener.php

```diff
@@ -87,7 +87,7 @@ public function publishRelatedModels($action = null, $entity = null)
         foreach ($models as $name => $association) {
             list(, $associationName) = pluginSplit($association->name());
             $viewVar = Inflector::variable($associationName);
-            if (array_key_exists($viewVar, $controller->viewVars)) {
+            if (array_key_exists($viewVar, (array)$controller->viewVars)) {
```

Comment:
See for yourself in an IDE, viewVars can be arrray or \ArrayObject. Thus the recommended cast.